##Импорт библиотек и загрузка данных

In [ ]:
# из библиотеки google.colab импортируем класс files
from google.colab import files

# создаем объект этого класса, применяем метод .upload()
uploaded = files.upload()

Saving data_with_coords_21_10_2023.csv to data_with_coords_21_10_2023.csv


In [ ]:
import pandas as pd
import seaborn as sns

In [ ]:
df_original = pd.read_csv("train.csv", sep=",")
df = pd.read_csv("updated_data_with_coords_22_10_2023.csv", sep=",")

##Парсер

In [ ]:
#Строки, в которых координаты не заданы
df_with_empty_lat = df_original[df_original['lat'].isna()]
df_with_empty_lat

Unnamed: 0      id  atm_group                                 address  \
40            40  4510.0      496.5  PER. BAZARNY 4            SPASSK-DALNY   
150          150   347.0     1022.0  ABB 6B NAB-CHELNINSKIJ    NAB.CHELNY     
182          182  1591.0     1942.0  16A BEREJKOVSK NAB        MOSCOW         
192          192   440.0     1942.0  113 MIRA AV.              TOGLIATTI      
214          214   556.0     1942.0  133A MOZHAYSKOE H/W.      MOSCOW         
...          ...     ...        ...                                     ...   
6158        6158  3751.0     8083.0  SEVASTOPOLSKIY ,43A.2  Magadan           
6166        6166  3794.0     8083.0  Universitetskiy 105    Volgograd         
6198        6198  4038.0     8083.0  KRONDSHTAT. ST., 113   Vyazma            
6222        6222  4209.0     8083.0  KSK 5 MKR, 29          CHita             
6247        6247  4354.0     8083.0  MOZHAYSKOE SH.,80A     Odintsovo         

     address_rus  lat  long    target  
40           NaN  NaN   NaN  0.039191  
150          NaN  NaN   NaN -0.145001  
182          NaN  NaN   NaN  0.080034  
192          NaN  NaN   NaN -0.082956  
214          NaN  NaN   NaN  0.061517  
...          ...  ...   ...       ...  
6158         NaN  NaN   NaN  0.040149  
6166         NaN  NaN   NaN  0.102689  
6198         NaN  NaN   NaN  0.067411  
6222         NaN  NaN   NaN  0.151250  
6247         NaN  NaN   NaN  0.141763  

[304 rows x 8 columns]

In [ ]:
from dadata import Dadata
#token = "602bff1c26954e118781e9a109d00ab6a6de0847"
#secret = "d2fa726a212448fb642337290f59605cf94f51ca"

dadata = Dadata(token, secret)

count = 0
for index, row in df_with_empty_lat.iterrows():
    result = dadata.clean("address", row['address'])
    #result = dadata.suggest("address", row['address'])
    '''
    print(f"Index: {index}")
    print(f"{row}\n")
    print(row['address'])
    print(result['geo_lat'])
    '''
    df.at[index, 'area_fias_id'] = result['area_fias_id']
    df.at[index, 'area_with_type'] = result['area_with_type']
    df.at[index, 'federal_district'] = result['federal_district']
    df.at[index, 'lat'] = result['geo_lat']
    df.at[index, 'long'] = result['geo_lon']
    df.at[index, 'capital_marker'] = result['capital_marker']
    df.at[index, 'city_with_type'] = result['city_with_type']
    df.at[index, 'city_fias_id'] = result['city_fias_id']
    df.at[index, 'city_district_with_type'] = result['city_district_with_type']
    df.at[index, 'city_district_fias_id'] = result['city_district_fias_id']
    df.at[index, 'fias_id'] = result['fias_id']
    df.at[index, 'fias_level'] = result['fias_level']
    df.at[index, 'region_with_type'] = result['region_with_type']
    df.at[index, 'region_fias_id'] = result['region_fias_id']
    df.at[index, 'settlement_with_type'] = result['settlement_with_type']
    df.at[index, 'settlement_fias_id'] = result['settlement_fias_id']
    df.at[index, 'street_with_type'] = result['street_with_type']
    df.at[index, 'street_fias_id'] = result['street_fias_id']
    df.at[index, 'city_area'] = result['city_area']
    df.at[index, 'street'] = result['street']
    df.at[index, 'house_fias_id'] = result['house_fias_id']
    df.at[index, 'address_rus'] = result['result']
    df.at[index, 'house'] = result['house']
    print(index, row['address'], result['result'])
    if result['metro']:
      df.at[index, 'metro'] = result['metro'][0]['distance']
    #print(count)

    '''
    Нет данных 'city_lat', 'city_lon', 'region_lat', 'region_lon', 'street_lat', 'street_lon'
     'area_lat', 'area_lon', 'settlement_lat',
       'settlement_lon', 'metro_station_name_1', 'metro_line_name_1',
       'metro_distance_1', 'metro_station_name_2', 'metro_line_name_2',
       'metro_distance_2', 'metro_station_name_3', 'metro_line_name_3',
       'metro_distance_3'

     'area_fias_id', 'area_with_type', 'city_with_type',
       'city_fias_id', 'federal_district', 'capital_marker', 'fias_id',
       'fias_level', 'region_with_type', 'region_fias_id',
       'settlement_with_type', 'settlement_fias_id', 'street_with_type',
       'street_fias_id', 'city_lat', 'city_lon', 'region_lat', 'region_lon',
       'street_lat', 'street_lon', 'city_area', 'city_district_fias_id',
       'city_district_with_type', 'area_lat', 'area_lon', 'settlement_lat',
       'settlement_lon', 'metro_station_name_1', 'metro_line_name_1',
       'metro_distance_1', 'metro_station_name_2', 'metro_line_name_2',
       'metro_distance_2', 'metro_station_name_3', 'metro_line_name_3',
       'metro_distance_3'],
     '''

    # Add more fields as needed

##Обработано вручную

1122 PAVELETSKAYA PL.,2,2   Moskva          г Москва
'Москва, Павелецкая площадь, 2, 2'

641 79, UL. INTERNACIONAL'NAY USOLYE-SIBIR
1245 ALTYF. SH. 22          Moskva
'г Москва Алтуфьевское ш., 22'
1744 UL. CSHORSA-8 V           BELGOROD     г Белгород
1826 D. 42, LIT. D, PR-KT 25 O GATCHINA G
'Гатчина, проспект 25 октября, д 42, лит д'
1854 D. 63, UL. KRASNOGEROISKA IZHEVSK G
'Ижевск, ул. Красногеройская, д 63'
2126 D. 51, LIT. A, UL. ADMIRA ASTRAKHAN G
'Астрахань, ул Адмирала, д 51, лит А'
2338 MKRN.INTERNATSOIN, 7A  Sayanogorsk
'Саяногорск, мкрн. Интернациональный, 7А'
2360 GER. TIKHOOKEAN., 5A   Vladivostok
Владивосток, Героев тихоокеанцев 5а
2369 SUKHOVA, 55B           Ussuriysk
'Уссурийск, Сухова, 55В'
2465 TURGENEVSKOE S., 27    KRASHODAR
Краснодар, Тургеневское, 27
2607 HO CHI MIHN AVE, 19       ULYANOVSK
Ульяновск, пр. Хо ши мина, 19
2703 7 NOSOVIKHINSKOE H.W.     REUTOV
2966 UL. KOROLEVA-9 A          BELGOROD
Белгород, ул Королева, 9А
2980 PAVLOVSK. TR., 251V       BARNAUL
Барнаул, Павловский тракт , 251В
3107 D. 105/107, UL. KOMMUNIST VOLSK G  
Вольск, ул Коммунистическая, д. 105/107
3260 D. 137T, PR-KT B.HMEL'NIC BELGOROD G
Белгород, пр. б. хмельницкого д. 137т
3273 D. 11B, UL. REVOLUCIONNAY TOLYATTI G
3274 D. 11B, UL. REVOLUCIONNAY TOLYATTI G  
3275 D. 11B, UL. REVOLUCIONNAY TOLYATTI G   
3276 D. 11B, UL. REVOLUCIONNAY TOLYATTI G   
3365 D. 58, LIT. A, SH. GOSTIL PETERGOF G  
3397 D. 4, UL. BLAGOVESHENSKAY VOLOGDA G  
3614 AVIAITSIONNAYA, 12     Komsomolsk-31
3654 KRANDSHTAT. BR., 37B   Moskva
3710 GOLOVATOGO, 313        Kemerovo
3918 ABB 111 KRASNOARMEJSK.    JOSHKAR-OLA
4206 MAJOROVA, 18              OMSK
4422 D. 123, UL. KRASNOARMEISK BLAGOVESHCHE
4449 D. 123, UL. KRASNOARMEISK BLAGOVESHCHE
4481 D. 23A, UL. KRASNOZNAMENN KRASNOZNAMEN
4604 D. 120, LIT. 1, NAB. OBVO SANKT-PETERB
4622 D. 17/1, NAB. USHAKOVSKAY SANKT-PETERB
4774 D. 11, LIT. A, SH. VYBORG SANKT-PETERB
4858 KRIZHIZHANOV,24/35/6   Moskva  
4863 CHERNYAKOVSK., 76, 76B Kaliningrad
4981 MOSKOVSKIY, 171a       Sankt-Peterburg
5019 PR.100LET VLADIV,88    VLADIVOSTOK
5622 D. 43, PR-KT 60 LET OKTYA KRASNOYARSK
5918 D. 27A, UL. ELEKTROVOZNAY VORONEZH G
6146 78 DOBROVOL. BR.,11    Krasnoyarsk
6198 KRONDSHTAT. ST., 113   Vyazma

In [ ]:
from dadata import Dadata
#token = "bd862b5111b50c71399a8edd451392e4adb028de"
#secret = "d5eac715841416f41057e34da941b53282af2f6c"
token = "602bff1c26954e118781e9a109d00ab6a6de0847"
secret = "d2fa726a212448fb642337290f59605cf94f51ca"

#secret = '2f4dac20b59dd41fe71a41c2ed81a61653331fa6'
#tocken = '69e4a2b28ce04367bb4db9b6595d5b5ae1367f0b'

dadata = Dadata(token, secret)

address = 'D. 23A, UL. KRASNOZNAMENN KRASNOZNAMEN	'
address = 'Краснознаменск, Краснознаменная улица, д.23А '
address = 'Пенза, Терновская,1'
address = 'Краснодар, Тургеневское, 27'
address = 'MOZHAYSKOE SH.,80A Odintsovo'
address = 'PER. BAZARNY 4            SPASSK-DALNY'
address = 'Москва, Бережковская набережная, 16А ' #16A BEREJKOVSK NAB        MOSCOW
address = 'Москва, Павелецкая площадь, 2, 2' #1122 PAVELETSKAYA PL.,2,2   Moskva          г Москва
address = "Ул Фаворского г Санкт-Петербург, д 12, лит 5" #602 D. 12, LIT. V, UL. FAVORS SANKT-PETERB г Санкт-Петербург
address = "г Усолье-Сибирское, ул. Интернациональная, д 79" #641 79, UL. INTERNACIONAL'NAY USOLYE-SIBIR
address = 'г Москва Алтуфьевское ш., 22' #1245 ALTYF. SH. 22          Moskva          г Москва
address = 'ул. Щорса, 8В, Белгород' #1744 UL. CSHORSA-8 V           BELGOROD     г Белгород
address = 'Гатчина, проспект 25 октября, д 42, лит д' #1826 D. 42, LIT. D, PR-KT 25 O GATCHINA G
address = 'Ижевск, ул. Красногеройская, д 63' #1854 D. 63, UL. KRASNOGEROISKA IZHEVSK G
address = 'Астрахань, ул Адмирала, д 51, лит А' #2126 D. 51, LIT. A, UL. ADMIRA ASTRAKHAN G
address = 'Саяногорск, мкрн. Интернациональный, 7А' #2338 MKRN.INTERNATSOIN, 7A  Sayanogorsk
address = 'Владивосток, Героев тихоокеанцев 5а' #2360 GER. TIKHOOKEAN., 5A   Vladivostok
address = 'Уссурийск, Сухова, 55В'#2369 SUKHOVA, 55B           Ussuriysk
address = 'Краснодар, Тургеневское, 27' #2465 TURGENEVSKOE S., 27    KRASHODAR
address = 'Ульяновск, пр. Хо ши мина, 19'#2607 HO CHI MIHN AVE, 19       ULYANOVSK
address = 'Реутов, Носовихинское шоссе, 7'#2703 7 NOSOVIKHINSKOE H.W.     REUTOV
address = 'Белгород, ул Королева, 9А' #2966 UL. KOROLEVA-9 A          BELGOROD
address = 'Барнаул, Павловский тракт , 251В'#2980 PAVLOVSK. TR., 251V       BARNAUL
address = 'Вольск, ул Коммунистическая, д. 105/107'#3107 D. 105/107, UL. KOMMUNIST VOLSK G
address = 'Белгород, пр. б. хмельницкого д. 137т'#3260 D. 137T, PR-KT B.HMEL'NIC BELGOROD G
address = 'Тольятти, ул. Революционная, д.11В' #3273 D. 11B, UL. REVOLUCIONNAY TOLYATTI G
#3274 D. 11B, UL. REVOLUCIONNAY TOLYATTI G   Самарская обл, г Тольятти
#3275 D. 11B, UL. REVOLUCIONNAY TOLYATTI G   Самарская обл, г Тольятти
#3276 D. 11B, UL. REVOLUCIONNAY TOLYATTI G   Самарская обл, г Тольятти
address = 'Екатеринбург, ул. Линак, д.128 лит А' #3314 D. 128, LIT. A, UL. LUNAC EKATERINBURG г Екатеринбург
#3315 D. 128, LIT. A, UL. LUNAC EKATERINBURG г Екатеринбург
address = 'Петергоф, Гостилицкое шоссе, д 58, лит А' #3365 D. 58, LIT. A, SH. GOSTIL PETERGOF G
address = 'Вологда, ул. Благовещенская, д 4'#3397 D. 4, UL. BLAGOVESHENSKAY VOLOGDA G
address = 'тер. Комсомольск 31, ул. Авиационная, 12 '#3614 AVIAITSIONNAYA, 12     Komsomolsk-31
address = 'Москва, Кронштадтский бульвар, 37В'#3654 KRANDSHTAT. BR., 37B   Moskva
address = 'Новосибирск, метро пл Ленина'#3696 metro Pl.Lenina        Novosibirsk
address = 'Йошкар-Ола, ул Красноармейская, 111' #3918 ABB 111 KRASNOARMEJSK.    JOSHKAR-OLA
address = 'Омск, Майорова, 18'#4206 MAJOROVA, 18              OMSK
address = 'Благовещенское, ул Красноармейская, д 123 '#4422 D. 123, UL. KRASNOARMEISK BLAGOVESHCHE
address = 'Благовещенское, ул Красноармейская, д 123' #4449 D. 123, UL. KRASNOARMEISK BLAGOVESHCHE
address = 'Краснознаменск, ул. Краснознаменная, д.23А' #4481 D. 23A, UL. KRASNOZNAMENN KRASNOZNAMEN
address = 'Санкт-Петербург, Набережная Обводного канала, д.120, лит 1'#4604 D. 120, LIT. 1, NAB. OBVO SANKT-PETERB
address = 'Санкт-Петербург, Выборское шоссе, д 11, лит А'#4774 D. 11, LIT. A, SH. VYBORG SANKT-PETERB
address = 'Ростов-на-Дону, ул. Расноармейская, д.63'#4858 KRIZHIZHANOV,24/35/6   Moskva
address = 'Москва, ул. Кржижановского, 24/35 корпус 6' #4858 KRIZHIZHANOV,24/35/6   Moskva
address = 'Калининград, Черняховского, 76-76Б'#4863 CHERNYAKOVSK., 76, 76B Kaliningrad
address = 'Санкт-Петербург, ул. Московский, 171а' #4981 MOSKOVSKIY, 171a       Sankt-Peterburg
address = 'Владивосток, пр-т. 100-летия Владивостока, 88' #5019 PR.100LET VLADIV,88    VLADIVOSTOK
address = 'Владивосток, Тихоокеанский, 98А' #5295 98A TIKHOOKEANSKIY        VLADIVOSTOK
address = 'Красноярск, Проспект 60 лет Октября, д 43' #5622 D. 43, PR-KT 60 LET OKTYA KRASNOYARSK
address = 'Воронеж, ул. Электровозная, д.27А' #5918 D. 27A, UL. ELEKTROVOZNAY VORONEZH G
address = 'Красноярск, ул. 78 Добровольческой Бригады, д 11' #6146 78 DOBROVOL. BR.,11    Krasnoyarsk
address = 'Вязьма, Кронштадтская ул., 113' #6198 KRONDSHTAT. ST., 113   Vyazma
index = 6198
result = dadata.clean("address", address)
df.at[index, 'area_fias_id'] = result['area_fias_id']
df.at[index, 'area_with_type'] = result['area_with_type']
df.at[index, 'federal_district'] = result['federal_district']
df.at[index, 'lat'] = result['geo_lat']
df.at[index, 'long'] = result['geo_lon']
df.at[index, 'capital_marker'] = result['capital_marker']
df.at[index, 'city_with_type'] = result['city_with_type']
df.at[index, 'city_fias_id'] = result['city_fias_id']
df.at[index, 'city_district_with_type'] = result['city_district_with_type']
df.at[index, 'city_district_fias_id'] = result['city_district_fias_id']
df.at[index, 'fias_id'] = result['fias_id']
df.at[index, 'fias_level'] = result['fias_level']
df.at[index, 'region_with_type'] = result['region_with_type']
df.at[index, 'region_fias_id'] = result['region_fias_id']
df.at[index, 'settlement_with_type'] = result['settlement_with_type']
df.at[index, 'settlement_fias_id'] = result['settlement_fias_id']
df.at[index, 'street_with_type'] = result['street_with_type']
df.at[index, 'street_fias_id'] = result['street_fias_id']
df.at[index, 'city_area'] = result['city_area']
df.at[index, 'address_rus'] = result['result']
df.at[index, 'street'] = result['street']
df.at[index, 'house_fias_id'] = result['house_fias_id']
df.at[index, 'address_rus'] = result['result']
df.at[index, 'house'] = result['house']
print(address, result['result'])
if result['metro']:
  df.at[index, 'metro'] = result['metro'][0]['distance']


##На удаление


440 PRESNENSKAJA              MOSKVA
911 UL. SHIROKOVA V.F.        KOSTROMA G   г Кострома, пл Широкова В.Ф.
1106 Dmitriya Donskogo b-r, Moskva          г Москва, б-р Дмитрия Донского
498 PL. KOMENDANTSKAYA        SANKT-PETERB г Санкт-Петербург, пл Комендантская
534 UL. GOGOLYA               NOVOSIBIRSK  г Новосибирск, ул Гоголя
911 UL. SHIROKOVA V.F.        KOSTROMA G   г Кострома, пл Широкова В.Ф.
1106 Dmitriya Donskogo b-r, Moskva          г Москва, б-р Дмитрия Донского
1177 PARTIZANA ZHELEZNYAKA, KRASNOYARSK     г Красноярск, ул Партизана Железняка
1732 PRESNENSKAJA NABERE       MOSKVA       г Москва, наб Пресненская
2301 Proezd Bagrationovskij,Moscow  
2348 Ploshchad Paveletskaya Moskva  
2362 Marshala Poluboyarova, Moskva
2936 G. ELEC, UL. RADIOT       ELEC   
2937 PR.  ANDROPOVA            MOSKVA
2950 CHL ARTIL-SKAJA100V       CHELJABINSK
2955 SML UL. 25 SENTJABR       SMOLENSK
2956 SML UL. 25 SENTJABR       SMOLENSK
3000 UL. KAVALERIJSKAJA,       NOVOSIBIRSK
3544 Novocheremushkinskaya, Moskva   
3588 Novocheremushkinskaya, Moskva
3603 PARTIZANA ZHELEZNYAKA, Krasnoyarsk
4246 LENINGRADSKOE SH.,        MOSKVA
4261 G.KURSK, PR. HRUCSH
4307 SH. ENTUZIASTOV  
4411 D. 27A                    BELORECHENSK
5965 UL. ANTONOVA              PENZA G

##Под вопросом

Не могу распознать
3314 D. 128, LIT. A, UL. LUNAC EKATERINBURG
Нет адреса Екатеринбург, ул. Луначарского, д.128 лит А. Есть тольок без литеры Есть банкомат ВТБ
3315 D. 128, LIT. A, UL. LUNAC EKATERINBURG г Екатеринбург
3710 GOLOVATOGO, 313        Kemerovo  - нет такого адреса в Кемерово, есть в Краснодаре
4152 64 SCHERTSA STR           BELGOROD
4824 RASNOARMEYSK.,63       Rostov-na-Donu
Нет такой улицы. Возможно Красноармейск, но там только банкомат от промсвязи, а должен быть Росбанка

5295 98A TIKHOOKEANSKIY        VLADIVOSTOK  г Владивосток
5296 98A TIKHOOKEANSKIY        VLADIVOSTOK  г Владивосток
5297 98A TIKHOOKEANSKIY        VLADIVOSTOK  г Владивосток
Нет такого адреса, только Океанский проспект, 98а отделение Тихоокеанский, но atm относится к альфабанку
6158 SEVASTOPOLSKIY ,43A.2  Magadan


5509 G.TJUMEN UL.RESPUBL       TJUMEN

Не дом
3696 metro Pl.Lenina        Novosibirsk
г Новосибирск, пл им. Ленина